network/src/router.rs

```diff
@@ -266,6 +353,30 @@ enum StatementTrace {
 	Invalid(ValidatorIndex, Hash),
 }
 
+/// Helper for deferring erasure-encoded chunks whose associated candidate block is unknown.
+struct DeferredChunks {
+	deferred: HashMap<Hash, Vec<ErasureChunk>>,
+}
+
+impl DeferredChunks {
+	fn new() -> Self {
+		DeferredChunks {
+			deferred: HashMap::new(),
+		}
+	}
+
+	fn push(&mut self, chunk: ErasureChunk) {
+		self.deferred.entry(chunk.candidate_hash).or_insert_with(Vec::new).push(chunk);
+	}
+
+	fn get_deferred(&mut self, hash: &Hash) -> Vec<ErasureChunk> {
```

Comment:
`get` implies borrowing - implementation is `take`ing.
```suggestion
	fn take_deferred(&mut self, hash: &Hash) -> Vec<ErasureChunk> {
```